prow/crier/metrics.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package crier reports finished prowjob status to git providers.
+package crier
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// Prometheus Metrics
+var (
+	// Use Fibonacci to get a good distribution. We care about report delayed
+	// for minutes, and even up to an hour.
+	reportTimeBuckets = []float64{1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181}
+
+	crierMetrics = struct {
+		reportTime *prometheus.HistogramVec
+	}{
+		reportTime: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "report_time",
+			Help:    "Histogram of time spent reporting, calculated by the time difference between job completion and end of reporting.",
+			Buckets: reportTimeBuckets,
+		}, []string{
+			"org",
+			"repo",
+			"jobtype",
+			"jobname",
+			"reporter",
```

Comment:
We're already seeing problems with excessive label cardinality with `prowjob_state_transitions`, this seems to use a similar label set, but also has 18 histogram counter buckets per label combination. I think this is going to cause even larger problems as a result.

Reply:
yeah, the same prowjob may report this multiple times, so this might be even worse than `prowjob_state_transitions`, I'll think about how to make this smaller. Please also update here if there is a better idea

Reply:
I'd suggest getting rid of all the labels except `reporter` and/or using fewer histogram buckets.  I don't think we really care about the `jobname` for this metric. The `org`, `repo`, and `jobtype` could be a bit interesting, but are not critical. I think we mainly just need this metric scoped to the `reporter`.

Reply:
agreed, currently reporter should be sufficient. How about add a log with all these fields for easier debugging if ever needed?

Reply:
done